Code/Framework/AzToolsFramework/Tests/Prefab/Instance/InstanceDeserializationTests.cpp

```diff
@@ -0,0 +1,205 @@
+                entity->Init();
+                entity->Activate();
+                return true;
+            });
+
+        return AZStd::make_pair<InstanceUniquePointer, InstanceUniquePointer>(AZStd::move(createdPrefab), AZStd::move(instantiatedPrefab));
+    }
+
+    TEST_F(InstanceDeserializationTest, ReloadInstanceUponComponentUpdate)
+    {
+        AZ::Entity* entity1 = CreateEntity("Entity1",false);
+        AzToolsFramework::Components::TransformComponent* transformComponent = aznew AzToolsFramework::Components::TransformComponent;
+        transformComponent->SetWorldTranslation(AZ::Vector3(10.0, 0, 0));
+        entity1->AddComponent(transformComponent);
+
+        AZ::Entity* entity2 = CreateEntity("Entity2", false);
+
+        AZStd::pair<InstanceUniquePointer, InstanceUniquePointer> prefabInstances =
+            SetupPrefabInstances(AzToolsFramework::EntityList{ entity1, entity2}, m_prefabSystemComponent);
+        InstanceUniquePointer createdPrefab = AZStd::move(prefabInstances.first);
+        InstanceUniquePointer instantiatedPrefab = AZStd::move(prefabInstances.second);
+
+        createdPrefab->GetEntities(
+            [](const AZStd::unique_ptr<AZ::Entity>& entity)
+            {
+                if (entity->GetName() == "Entity1")
+                {
+                    // Activate the entity to access the transform interface and use it to modify the transform component.
+                    entity->Init();
+                    entity->Activate();
+                    AZ::TransformBus::Event(entity->GetId(), &AZ::TransformInterface::SetWorldX, 20.0f);
+                }
+                return true;
+            });
+        
+        
+        GenerateDomAndReloadInstantiatedPrefab(createdPrefab, instantiatedPrefab);
+
+        instantiatedPrefab->GetEntities(
+            [](const AZStd::unique_ptr<AZ::Entity>& entity)
+            {
+                if (entity->GetName() == "Entity2")
+                {
+                    // Since we didn't touch entity2 in createdPrefab, it should remain untouched in instantiatedPrefab and thus retain its
+                    // active state.
+                    EXPECT_EQ(entity->GetState(), AZ::Entity::State::Active);
+                }
+                else
+                {
+                    // Validate that the entity is in 'constructed' state, which indicates that it got reloaded.
+                    EXPECT_EQ(entity->GetState(), AZ::Entity::State::Constructed);
+                    AZStd::vector<AZ::Component*> entity1Components = entity->GetComponents();
+                    EXPECT_TRUE(entity1Components.size() == 1);
+                    AzToolsFramework::Components::TransformComponent* transformComponentInInstantiatedPrefab =
+                        reinterpret_cast<AzToolsFramework::Components::TransformComponent*>(entity1Components.front());
+                    EXPECT_TRUE(transformComponentInInstantiatedPrefab != nullptr);
```

Comment:
repeat: This should should use EXPECT_NE
```suggestion
                    EXPECT_NE(nullptr, transformComponentInInstantiatedPrefab);
```